stdlib/Test/src/Test.jl

```diff
@@ -85,8 +85,9 @@ struct Pass <: Result
     orig_expr
     data
     value
-    function Pass(test_type::Symbol, orig_expr, data, thrown)
-        return new(test_type, orig_expr, data, thrown isa String ? "String" : thrown)
+    source::Union{Nothing,LineNumberNode}
+    function Pass(test_type::Symbol, orig_expr, data, thrown, source)
```

Comment:
A package (Jute) was using the constructor directly in its tests. Should we add a default value (nothing)?

Reply:
Something like this?

```Julia
Pass(test_type::Symbol, orig_expr, data, thrown) = Pass(test_type, orig_expr, data, thrown, nothing)
```

Does it make more sense for Jute to update its use of the constructor if it's an internal API? Happy either way just wanted to check.

Reply:
I think that makes sense, we generally try not to break packages unnecessarily. That can equivalently be written more concisely as:
```
function Pass(test_type::Symbol, orig_expr, data, thrown, source=nothing)
```

Reply:
Sounds good - I won't be able to get to it for a few weeks as I'm away from my computer but will do it then.

I'm not quite sure how your suggestions works, doesn't that need an `end` to work and you'd need to pass the arguments to `Pass`?

Reply:
finally in a position to look at this (sorry for delay) but think its been done now (#41424 ), so will resolve this comment.